File: python_examples/hello_plugin.py
from typing import Sequence, override
import mobase
import sys

from PyQt6.QtCore import qInfo

class HelloWorldPlugin(mobase.IPlugin):
    _organizer: mobase.IOrganizer
    
    def __init__(self):
        print("Initializing HelloWorldPlugin")
        super().__init__()

    @override
    def init(self, organizer: mobase.IOrganizer) -> bool:
        self._organizer = organizer
        print("Print logs at DEBUG level")
        print("Errors go to ERROR level", file=sys.stderr)
        qInfo("This one way to put things into the logs at INFO level")
        return True

    @override
    def name(self):
        return "Python: Hello World Plugin"

    @override
    def author(self):
        return "Your Name"

    @override
    def description(self):
        return "A simple plugin for Mod Organizer 2"

    @override
    def version(self):
        return mobase.VersionInfo(1, 0, 0, mobase.ReleaseType.FINAL)

    @override
    def settings(self) -> Sequence[mobase.PluginSetting]:
        return []
        
def createPlugin() -> mobase.IPlugin:
    return HelloWorldPlugin()

File: cpp_examples/hello_plugin/src/hello_plugin.cpp
#include "hello_plugin.h"

// Constructor and Destructor
HelloPlugin::HelloPlugin() : m_Organizer(nullptr) {}

HelloPlugin::~HelloPlugin() {}

// IPlugin interface
bool HelloPlugin::init(IOrganizer* organizer) {
    m_Organizer = organizer;
    MOBase::log::info("Hello, Plugin!");
    return true;
}

QString HelloPlugin::name() const { return "HelloPlugin"; }

QString HelloPlugin::author() const { return "Your Name"; }

QString HelloPlugin::description() const { return "A simple plugin for Mod Organizer."; }

VersionInfo HelloPlugin::version() const { return VersionInfo(1, 0, 0); }

QList<PluginSetting> HelloPlugin::settings() const {
    return {};  // No settings for this example
}
